### PR TITLE
soroban-rpc: Obtain fee configuration from the ledger

### LINKED ISSUE #665: soroban-rpc
### What ? 
Read the config settings ledger keys when calculating fees

### PR DESCRIPTION
**Pull request by 2opremio**
### What

Obtain the configuration for the fee estimation of `simulateTransaction` from the ledger instead of hardcoding it.

### Why

Close #665 

### Known limitations

Obtaining 4 ledger entries for each `simulateTransaction` call will reduce performance. We need a benchmark to measure the impact.

### CODE REVIEW

File: cmd/soroban-rpc/lib/preflight/src/fees.rs [hunk 1]
```diff
@@ -140,23 +141,74 @@ fn calculate_soroban_resources(
...
-    //       Specifically, we need to derive it from ConfigSettingContractComputeV0 which can
-    //       be retrieved using the ConfigSetting/CONFIG_SETTING_CONTRACT_COMPUTE_V0.
+fn get_configuration_setting(
+    ledger_storage: &ledger_storage::LedgerStorage,
+    setting_id: ConfigSettingId,
+) -> Result<ConfigSettingEntry, Box<dyn error::Error>> {
+    let key = LedgerKey::ConfigSetting(LedgerKeyConfigSetting {
+        config_setting_id: setting_id,
+    });
+    match ledger_storage.get(&key)? {
```
Comment (paulbellamy) [inline cmd/soroban-rpc/lib/preflight/src/fees.rs:151]: We could refactor ledger_storage so `get` can fetch multiple keys at the same time. Might reduce the go<->rust round-trips. But we should probably only do that if it proves to be an issue.

File: cmd/soroban-rpc/lib/preflight/src/fees.rs [hunk 2]
```diff
@@ -140,23 +141,74 @@ fn calculate_soroban_resources(
...
+        )
+        .into()),
+    }
+}
+
+fn get_fee_configuration(
+    ledger_storage: &ledger_storage::LedgerStorage,
+) -> Result<FeeConfiguration, Box<dyn error::Error>> {
+    // TODO: consider caching these entries. In theory, they can change at any given ledger
+    //       (making caching ineffective) but we can maybe relax that requirement?
```
Comment (paulbellamy) [inline cmd/soroban-rpc/lib/preflight/src/fees.rs:168]: If they don't change frequently, but we need immediate results when they *do* change, a write-through cache might be the right choice here.

File: cmd/soroban-rpc/lib/preflight/src/fees.rs (hunk 1 above)
Comment (paulbellamy) [reply cmd/soroban-rpc/lib/preflight/src/fees.rs:151]: Do we have a good way to benchmark this given the rust/go boundary stuff? would a go benchmark be reliable?

**Review by paulbellamy — APPROVED** [review]

File: cmd/soroban-rpc/lib/preflight/src/fees.rs (hunk 2 above)
Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/fees.rs:168]: That's a really good idea

Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/fees.rs:168]: I will add a comment about it

**Pull request merged**